lib/logstash/codecs/retriggerable_task.rb

```diff
@@ -0,0 +1,90 @@
+require "concurrent"
+
+module LogStash module Codecs class RetriggerableTask
+  class DummyListener
+    def timeout
+      STDOUT.puts "..... task timed out ....."
+    end
+  end
+
+  attr_reader :thread
+
+  def initialize(delay, listener = DummyListener.new)
+    @delay = calculate_delay(delay)
+    @listener = listener
+    @counter = Concurrent::AtomicFixnum.new(0 + @delay)
+    @stopped = Concurrent::AtomicBoolean.new(false)
+    @semaphore = Concurrent::Semaphore.new(1)
+    @semaphore.drain_permits
+  end
+
+  def retrigger
+    return if stopped?
+    if executing?
+      @semaphore.acquire
+    end
+
+    if pending?
+      reset_counter
+    else
+      start
+    end
+  end
+
+  def close
+    @stopped.make_true
+  end
+
+  def counter
+    @counter.value
+  end
+
+  def executing?
+    running? && counter < 1
+  end
+
+  def pending?
+    running? && counter > 0
+  end
+
+  private
+
+  def calculate_delay(value)
+    # in multiples of 0.25 seconds
+    return 1 if value < 0.25
+    (value / 0.25).floor
```

Comment:
Can we move this 0.25 value to a constant? I understand this is the `sleep` step for the thread.